.github/actions/update-charm-pins/action.yaml

```diff
@@ -0,0 +1,29 @@
+---
+name: Update Charm Pins
+description: Updates pinned versions of external charms we use to test our changes against to prevent regressions
+author: Dima Tisnek <dimaqq@gmail.com>
+branding:
+  icon: activity
+  color: orange
+
+inputs:
+  workflows:
+    description: Whitespace-separated paths to the local workflow file, relative to repository root
+    required: true
+  gh-pat:
+    description: Personal access token to check out external repos from GitHub
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - uses: actions/setup-python@v5
+      with:
+        python-version: "3.12"
+
+    - run: python -m pip install ruamel.yaml==0.18.6 httpx==0.27.0
```

Comment:
Do you know if dependabot is smart enough to find these? If not, then it's probably better to have a requirements file and install from that.